### PR TITLE
Pages are now stable after rotation/resize.

### LINKED ISSUE #642: PageableList loses position after rotation
Repro:

Run widgets/pageable_list.dart on your phone
- Scroll to Card 4.
- Rotate the phone 90 degrees and back

The list ends up showing the wrong card that's half scrolled.

### PR DESCRIPTION
**Pull request by jimbeveridge**
Fixes #642.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 1]
```diff
@@ -751,13 +753,21 @@ class PageableListState<T> extends ScrollableListState<T, PageableList<T>> {
...
   void _notifyPageChanged(_) {
+    _currentPage = (scrollOffset / config.itemExtent).floor() % itemCount;
     if (config.onPageChanged != null)
       config.onPageChanged(currentPage);
   }
 
+  void _updateScrollBehavior() {
+    // Handle rotation/resizing
+    var newScrollOffset = config.itemExtent * _currentPage;
+    scrollTo(newScrollOffset, duration: null);
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/scrollable.dart]: You can leave off the `duration: null`.  The default is `null`.

**Comment by abarth** [comment]
LGTM

Thanks for the patch!  (Ideally you would add a test, but it's fine if you'd rather not.)

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 2]
```diff
@@ -751,13 +753,21 @@ class PageableListState<T> extends ScrollableListState<T, PageableList<T>> {
...
 
   void _notifyPageChanged(_) {
+    _currentPage = (scrollOffset / config.itemExtent).floor() % itemCount;
     if (config.onPageChanged != null)
       config.onPageChanged(currentPage);
   }
 
+  void _updateScrollBehavior() {
+    // Handle rotation/resizing
+    var newScrollOffset = config.itemExtent * _currentPage;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scrollable.dart]: `double`, not `var`

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 3]
```diff
@@ -751,13 +753,21 @@ class PageableListState<T> extends ScrollableListState<T, PageableList<T>> {
...
-  int get currentPage => (scrollOffset / config.itemExtent).floor() % itemCount;
+  int get currentPage => _currentPage;
 
   void _notifyPageChanged(_) {
+    _currentPage = (scrollOffset / config.itemExtent).floor() % itemCount;
     if (config.onPageChanged != null)
       config.onPageChanged(currentPage);
   }
 
+  void _updateScrollBehavior() {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scrollable.dart]: why not in `_handleSizeChanged` ? `_updateScrollBehavior` is called from build, which doesn't seem like a good time to be calling scrollTo().

**Comment by Hixie** [comment]
Another way to fix this would be to keep track of the scroll offset as a fraction rather than a number of pixels. That might handle cases like rotating in the middle of a swipe better.

**Comment by jimbeveridge** [comment]
Abandoning this change because Hans is rewriting the classes to use fractional offsets as described in the code review comments.

**Pull request closed without merging**